File: backend/GraficaModerna.Application/Interfaces/ICartService.cs
using GraficaModerna.Application.DTOs;

namespace GraficaModerna.Application.Interfaces;

public interface ICartService
{
    Task<CartDto> GetCartAsync(string userId);
    Task AddItemAsync(string userId, AddToCartDto dto);
    Task RemoveItemAsync(string userId, Guid cartItemId);
    Task ClearCartAsync(string userId);
}

File: grafica-moderna/app/Application/ICartService.php
<?php

namespace App\Application\Interfaces;

interface ICartService
{
    public function getCart(string $userId): object;

    public function addItem(string $userId, array $dto): void;

    public function updateItemQuantity(string $userId, string $cartItemId, int $quantity): void;

    public function removeItem(string $userId, string $cartItemId): void;

    public function clearCart(string $userId): void;
}